Allow parameterizing autosign and modulepath for puppet master

claytono opened the pull request:
Allow overriding arguments to the `puppet::master` class by looking up the arguments in hiera, and falling back to the previous defaults if nothing is found.

claytono committed:
Allow parameterizing autosign and modulepath for puppet master

claytono committed:
Moved collectd class into monitoring server profile.

Collectd is logically part of the graphite configuration, since it
feeds into graphite/carbon and doesn't work without it being installed.
This moves the collectd class declaration into the monitoring profile.
so that collectd will only be installed if monitoring is enabled.

markvoelker commented:
Two bits here:

1.)  Looks like a second commit snuck in here from your other pull request.
2.)  Could you target this at Icehouse and cherry pick back to havana?

Otherwise I think is ok; I'll take it for a test run shortly.

claytono committed:
Remove NTP and naginator from coi base

We do this ourselves, so we don't need this anymore

claytono committed:
Fixed typo

claytono committed:
Use hiera param lookup for puppet::master class

This code appears to have been written for pre 3.0 puppet, which didn't
have automatic parameter lookup.  Moved the puppet::master class
inclusion to the include instead of class model so that we can have it
declared in more than one place without errors.
